frontend/device/kindle/device.lua

```diff
@@ -165,6 +165,35 @@ function Kindle:intoScreenSaver()
     if self:supportsScreensaver() then
         -- NOTE: Meaning this is not a SO device ;)
         if self.screen_saver_mode == false then
+            -- NOTE: Pilefered from Device:onPowerEvent @ frontend/device/generic/device.lua
+            -- Mostly always suspend in Portrait/Inverted Portrait mode...
+            -- ... except when we just show an InfoMessage or when the screensaver
+            -- is disabled, as it plays badly with Landscape mode (c.f., #4098 and #5290)
+            local screensaver_type = G_reader_settings:readSetting("screensaver_type")
+            if screensaver_type ~= "message" and screensaver_type ~= "disable" then
+                self.orig_rotation_mode = self.screen:getRotationMode()
+                -- Leave Portrait & Inverted Portrait alone, that works just fine.
+                if bit.band(self.orig_rotation_mode, 1) == 1 then
+                    -- i.e., only switch to Portrait if we're currently in *any* Landscape orientation (odd number)
+                    self.screen:setRotationMode(0)
+                else
+                    self.orig_rotation_mode = nil
+                end
+
+                -- On eInk, if we're using a screensaver mode that shows an image,
+                -- flash the screen to white first, to eliminate ghosting.
+                if self:hasEinkScreen() and
```

Comment:
arent all kindles (supported by koreader) Eink?

Reply:
Yep, left it in to make it extremely obvious this was a c/p ^^.